### PR TITLE
Replace JCenter with Maven Central.

### LINKED ISSUE #1055: Replace JCenter with Maven Central for Gradle Build
**Is your feature request related to a problem? Please describe.**

On February 3 2021, JFrog [announced](https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/ ) the shutdown of JCenter. Later on April 27 2021, an update was provided that the repository will only be read only and new package and versions are no longer accepted on JCenter.  This means we can no longer use JCenter for our central artifacts repository for dependency resolution.

**Describe the solution you'd like**
 Replace with Maven Central as per the [Gradle recommendation](https://blog.gradle.org/jcenter-shutdown)

### PR DESCRIPTION
**Pull request by adnapibar**
### Description

https://blog.gradle.org/jcenter-shutdown
 
### Issues Resolved
#1055 
 
### Check List
- [x] All tests pass
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Signed-off-by: Rabi Panda <adnapibar@gmail.com>

### CODE REVIEW

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -94,7 +94,8 @@ tasks.withType(JavaCompile).configureEach {
  *****************************************************************************/
 
 repositories {
-  jcenter()
+  mavenCentral()
+  gradlePluginPortal()
```
Comment (adnapibar) [inline buildSrc/build.gradle:98]: For plugins not in mavenCentral() as a fallback. The Gradle Plugin Portal implicitly mirrors JCenter currently.

**Comment by adnapibar** [comment]
start gradle check

**Review by saratvemulapalli — COMMENTED** [review]
On what dependencies of OpenSearch, we rely on jcenter() today? 

I do not see any specific dependency from jcenter() which cannot be fetched from mavenCentral() for OpenSearch, curious why it was added.

**Comment by adnapibar** [comment]
> On what dependencies of OpenSearch, we rely on jcenter() today?
> 
> I do not see any specific dependency from jcenter() which cannot be fetched from mavenCentral() for OpenSearch, curious why it was added.

All dependencies are resolved through jcenter today. This change removes that and uses mavenCentral instead.

**Review by saratvemulapalli — APPROVED** [review]

**Comment by saratvemulapalli** [comment]
> > On what dependencies of OpenSearch, we rely on jcenter() today?
> > I do not see any specific dependency from jcenter() which cannot be fetched from mavenCentral() for OpenSearch, curious why it was added.
> 
> All dependencies are resolved through jcenter today. This change removes that and uses mavenCentral instead.

Thanks for helping me understand. It would be nice to see this as description for the PR.

**Comment by adnapibar** [comment]
> Thanks for helping me understand. It would be nice to see this as description for the PR.

Thanks:) The linked issue has all the details.

**Pull request merged**